tools/clang/test/CodeGenHLSL/basic.hlsl

```diff
@@ -1,4 +1,5 @@
 // RUN: %dxc -E main -T ps_6_0 %s
+// TODO: No check lines found, we should update this
```

Comment:
There is no filecheck command on the RUN line. Shaders such as these, particularly those in the CodeGenHLSL directory, tend to be used for specific compilation tests that evaluate or processes the results in another way. 

For example, this shader is used in ValidationTest to generate IR output that is then manipulated to be faulty. There's no need for CHECK lines there.

Reply:
Part of the reason for me putting a TODO there, is to figure out how to convert the tests into something that can be run via LIT instead of a unit test wrapper. I can remove the comment, but the general idea is that these tests don't fit smoothly into the LLVM testing infrastructure and I wanted to note that for later investigation.

Reply:
CodeGenHLSL tests aren't ones that actually use the run lines or run under FileCheck generally.  So these are probably not ones you want to attempt to run via LIT.

So I agree with Chris that we should add FileCheck and CHECK lines to tests that are missing them under the HLSLFileCheck directory, I don't think that applies to files under the CodeGenHLSL or DXILValidation directories as a general rule.

However, the file does have a RUN line formatted as if it is run this way `// RUN: %dxc ...`.  So perhaps, for tests that do not run this way, and are never intended to do so, we should change the comment at the top to a less formal version that captures useful command line information, but doesn't imply that it runs under standard FileCheckerTest or LIT test context.

For some of these tests under CodeGenHLSL though, they could probably be made to run as normal FileCheck tests in the HLSLFileCheck directory, instead of custom groups under the CompilerTest.cpp, removing the single `CodeGenTest` or `CodeGenTestCheck` test cases from `CompilerTest.cpp` around lines 3240 to 3302.

Reply:
Ok, it would appear that this file is only used by ValidationTest.cpp, so it looks like basic.hlsl should be moved under DXILValidation and the //RUN line simply deleted.

Reply:
There appears to be a bunch of FileCheck-style tests in individual groups in DxilValidation.cpp as well.  So perhaps these should also be moved under `HLSLFileCheck/validation`, and the test groups that just invoke them with `TestCheck` removed.